File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> box = new Stack<>();
        
        for(int i=0;i<moves.length;i++){
            for(int j=0;j<board.length;j++){
                if(board[j][moves[i]-1]!=0){
                    int n=board[j][moves[i]-1];
                    board[j][moves[i]-1]=0;
                    if(box.size()>0){
                        if(n==box.peek()){
                            box.pop();
                            answer+=2;
                            break;
                        }
                    }
                    box.push(n);
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    basket=[]
    loc=[len(board)]*len(board)
    
    for c in range(len(board)):
        for r in range(len(board)):
            if board[r][c]!=0:
                loc[c]=r
                break
        
    for move in moves:
        if loc[move-1]>=len(board):
            continue
        elif board[loc[move-1]][move-1]!=0:
            if len(basket)>0 and basket[-1]==board[loc[move-1]][move-1]:
                basket.pop()
                answer+=2
            else:
                basket.append(board[loc[move-1]][move-1])
            loc[move-1]+=1
    
    return answer